src/com/goide/psi/impl/GoPsiImplUtil.java

```diff
@@ -1131,4 +1131,19 @@ public static boolean isUnaryBitAndExpression(@Nullable PsiElement parent) {
     PsiElement grandParent = parent != null ? parent.getParent() : null;
     return grandParent instanceof GoUnaryExpr && ((GoUnaryExpr)grandParent).getBitAnd() != null;
   }
+
+  @Nullable
+  public static PsiElement getBreakStatementOwner(@NotNull GoBreakStatement breakStmt) {
+    GoLabelRef label = breakStmt.getLabelRef();
+    if (label != null) {
+      return label.getReference().resolve();
+    }
+    GoCompositeElement breaksOutOf =
+      PsiTreeUtil.getParentOfType(breakStmt, GoSwitchStatement.class, GoForStatement.class, GoSelectStatement.class,
+                                  GoFunctionOrMethodDeclaration.class);
+    if (breaksOutOf instanceof GoFunctionOrMethodDeclaration) {
```

Comment:
Well, yes. We need to stop searching when we hit GoFunctionOrMethodDeclaration, no point looking higher up the tree. 

According to spec https://golang.org/ref/spec#Break_statements A "break" statement terminates execution of the innermost "for", "switch", or "select" statement within the same function.